src/components/toast/toast.test.js

```diff
@@ -0,0 +1,175 @@
+
+        toastComponent().should("have.css", "margin-right", `${margin}px`);
+      }
+    );
+
+    it("should render Toast component with className prop", () => {
+      CypressMountWithProviders(<ToastComponent className={testData} />);
+
+      toastComponent().should("have.class", testData);
+    });
+
+    it("should render Toast component with id prop", () => {
+      CypressMountWithProviders(<ToastComponent id={testData} />);
+
+      toastComponent().should("have.attr", "id", testData);
+    });
+
+    it("should render Toast component with data-component prop", () => {
+      CypressMountWithProviders(<ToastComponent data-component={testData} />);
+
+      getComponent(testData).should("exist");
+    });
+
+    it.each([
+      [TOAST_COLORS[0], colorTypes[0]],
+      [TOAST_COLORS[1], colorTypes[1]],
+      [TOAST_COLORS[2], colorTypes[2]],
+      [TOAST_COLORS[3], colorTypes[3]],
+    ])("should render Toast component with %s variant", (variant, color) => {
+      CypressMountWithProviders(<ToastComponent variant={variant} />);
+
+      toastComponent().should("have.css", "border").and("contain", color);
+    });
+
+    it.each([
+      [true, "be.visible"],
+      [false, "not.exist"],
+    ])(
+      "should render Toast component with open prop set to %s",
+      (bool, assertion) => {
+        CypressMountWithProviders(<ToastComponent open={bool} />);
+
+        toastComponent().should(assertion);
+      }
+    );
+
+    it.each([
+      [1, 1000],
+      [15, 15000],
+      [25, 25000],
+    ])(
+      "should render Toast component with timeout prop set to %s",
+      (timeout, numberForTick) => {
+        const now = new Date();
+        cy.clock(now);
+
+        CypressMountWithProviders(<ToastComponent timeout={timeout} />);
+
+        toastComponent().should("be.visible");
+        cy.tick(numberForTick);
+        toastComponent().should("not.exist");
+      }
+    );
+
+    it("should render Toast component with targetPortalId prop", () => {
+      CypressMountWithProviders(<ToastComponent targetPortalId={testData} />);
+
+      toastComponent()
+        .parent()
+        .parent()
+        .parent()
+        .parent()
+        .should("have.attr", "id", testData);
+    });
+
+    it("should render Toast component with maxWidth prop", () => {
+      CypressMountWithProviders(<ToastComponent maxWidth="250px" />);
+
+      toastComponent().should("have.css", "maxWidth", "250px");
+    });
+  });
+
+  describe("check events for Toast component", () => {
+    it("should call onDismiss callback when a click event is triggered", () => {
+      const callback = cy.stub();
+
+      CypressMountWithProviders(<ToastComponent onDismiss={callback} />);
+
+      closeIconButton()
+        .click()
+        .then(() => {
+          // eslint-disable-next-line no-unused-expressions
+          expect(callback).to.have.been.calledOnce;
+        });
+    });
+  });
```

Comment:
Should the onDismiss callback be tested when Esc is used to close component?

Reply:
agree, thx for pointing this, added a TC